src/helpers/ht_data.service.ts

```diff
@@ -14,13 +14,24 @@ export class HT_DataService {
     private session_key: string;
     private local_key: string;
 
-    constructor(username: string) {
-        username = username.toLowerCase();
-        this.setUsername(username);
-        let data = this.getData();
-        if (!data.saltBits) {
-            data.saltBits = JSON.stringify(HaventecCommon.generateSalt());
-            this.setData(data);
+    constructor(username?: string) {
+
+        if ( !username ) {
+            const haventec_username = HT_LocalStorage.getItem('haventec_username');
+            if ( haventec_username ) {
+                username = haventec_username.toString();
+            }
+        }
+
+        if ( username ) {
```

Comment:
Refactor into else statement since the false value of the first condition implies this condition

Reply:
The logic here is correct